docs/installation.md

```diff
@@ -218,19 +218,234 @@ On Linux, as an optional post-installation task, you may wish to setup the bot t
+
+```bash
+conda config --set auto_activate_base false
+```
+
+
+##### Change the channels with upgrades:
+
+Conda as a package manager can download new packages from "channels". The best developed Conda channel, is not the default channel and is called `conda-forge`. The code below switches to it.
+
+```bash    
+# adding forge
+conda config --env --add channels conda-forge
+
+# make it strict
+conda config --env --set channel_priority strict 	
+```
+
+You can check the status of conda with the following code.
+
+```bash
+conda info
+conda config --show channels
+conda config --show channel_priority
+```
+
+#### 3. Freqtrade Conda Environment
+
+Now you have conda, but only base environment,
+
+```bash# download freqtrade
+conda env list
 ```
 
+It is time to setup environment of the Freqtrade itself:
+
+
+Conda `create` command, installs all nested dependencies, for the selected libraries automatically
+
+general structure of installation command is:
+
+```bash
+conda create -n [name of the environment] [python version] [packages]
+```
+
+so it can be
+
+```bash
+conda create -n freqtrade-conda python=3.8 pandas numpy ta-lib git wheel virtualenv
+```
+
+or if you expect, to use later jupiter for [data-analysis](data-analysis.md), use
+
+```bash
+conda create -n freqtrade-jupyter-conda python=3.8 pandas numpy ta-lib git wheel virtualenv jupyter
+```
+
+the same works for [docker](docker.md), for spyder and other useful programs.
+
+Further read on the topic:
+
+https://towardsdatascience.com/a-guide-to-conda-environments-bc6180fc533?gi=1db972389cd1
+
+https://metager.de/meta/meta.ger3?eingabe=ardsdatascience+guide+to+conda+environment
+
+!!! NOTE:
+It may happen that creating a new Conda environment, populated with selected packages at the moment of creation, takes less time than installing a large, heavy dependent package into the old environment,
+takes less time than installing a large, heavy dependent package into the old environment.
+
+##### Enter/exit freqtrade-conda venv:
+
+to enter choosen conda environment
+
+```bash
+conda activate freqtrade-conda
+
+```
+
+to exit conda environment
+
+```bash
+conda deactivate						
+```
+
+#### 4. Freqtrade Instalation
+
+Within the newly created freqtrade-conda environment, download and install Freqtrade. To process --install command, may take a few minutes
+
+```bash
+# download freqtrade
+git clone https://github.com/freqtrade/freqtrade.git
+
+# enter downloaded directory 'freqtrade'
+cd freqtrade      
+
+# run setup
+./setup.sh --install                                        
+```
+
+Create virtual environment (yes, python3-venv environment, inside conda-environment), where Freqtrade program can run.
+
+Running python3-venv seems to be requirement of the Freqtrade program itself.
+
+```bash
+# create venv environment, inside hidden directory /freqtrede/.env
+python3 -m venv ./.env/
+
+# run the python3-venv environment
+source ./.env/bin/activate
+
```

Comment:
This is plain wrong. 
when you use a conda environment, you cannot use the setup.sh script.

You can however install directly via `pip install -r requirements.txt && pip install -e .`

Reply:
Yes, in conda, `./setup.sh --install` works perfectly,
provided documentation is provided right.

During `conda create -n ..` it is important to provide important packages which cover dependencies, 
then `pip install -r requirements.txt` will install nothing 
`pip install -e .` brings part of one dependency (out of dozens)

Also, according to the documentation, pip should be used as little as possible inside conda, which is mentioned in one of the `!!! Note`

With all respect, I would change here nothing.

Reply:
but there is no point in using the setup script when you use conda.

It's mixing conda with virtualenv, which the conda documentation recommends against.
I am sure it works fine for you, but we'll not have documentation that's contradicting to the official docs.

Reply:
`conda documentation recommends against` - could you double check it

Conda documentation nowhere is against mixing conda and venv.
There is offcial conda package for that
https://anaconda.org/anaconda/virtualenv

You know freqtrade program supposingly best, 
could you please explain, how  `./setup.sh --install` would go into conflict with conda?

To make it clear, approach which i've explained in the documentation :
1.Install Conda
2.Prepare Freqtrade-Conda environment, where most of the libraries are covered upfront
3.Turn on Freqtrade-Conda environment
4.Inside do  `./setup.sh --install`
5.Turn on p3-venv
6.Run freqtrade
7.Viola

Reply:
it can go into conflicts if you start combining packages from both sources.
Say you update numpy (installed via conda) - but you don't update pandas (installed via pip).

You can end up with bad dependencies, sometimes the dependency check will flat out fail.

soures for this:
[conda ](https://www.anaconda.com/blog/using-pip-in-a-conda-environment)
[stackoverflow](https://stackoverflow.com/questions/56134588/is-that-a-bad-idea-to-use-conda-and-pip-install-on-the-same-environment#:~:text=Don't%20mix%20conda%20install%20and%20pip%20install%20within%20conda%20environment.&text=Conda%20installs%20various%20(not%20only,virtualenv%20is%20one%20of%20them).)

I do have bad experiences with this (dependencies breaking) - so i will NOT document this as part of freqtrade unless anaconda RECOMMENDS doing this (which they don't).

---

If we recommend to use `./setup.sh --install` also with conda, then this whole guide is pointless and we can simply say something around "please install conda based on the installation guide - activate base, then run `./setup.sh --install`.

it's using a virtual env - there would be no point in creating a conda environment, you can create the virtual env out of base.

Reply:
With such a clear answer, if I understand it right, I have no choice but to abandon the path of including the conda path I invented in the official freqtrade documentation.

At the same time, it might be the right thing to do to completely remove the traces of Conda from the documentation. if the clearest way is not sufficient.

Is topic closed, or there is anything relevant to add?

============

In due time, I intend to take a look at the existing documentation, because of the internal problems:

docs/docker_quickstart.md
there are fundamental deficiencies, which does not allow to move installation according to the documentation.

docs/configuration.md
currently covers a broad spectrum of activities, but could be more structured for clarity.

Reply:
> With such a clear answer, if I understand it right, there is no choice, but to abandon the path of including the conda path, in the official freqtrade documentation, that has been invented and revealed above.

Well, conda itself says that this way can be problematic (see source above) - so as said, it CAN work - but it may also fail (especially if people think "oh, i got a conda instance, so letme install dependency X via conda" - after which you may end up with 2 (different) pandas versions installed for example). 

If you know what to do with conda, what to avoid, i'm sure it can work just fine - but keeping in mind that 80-90% of freqtrade users are non-technical, i see this as problematic - as it'll then be up to the freqtrade team to deal with issues of people which "follow the documentation - which suggested this - and now you tell me it's not working if i also do this what i found on ...".

most of what you did is fine - but i disagree on using the setup script, but instead installing freqtrade directly in the conda environment. This will cause less problem as it's not a virtualenv nested in a conda environment.
Technically, you could also create a virtualenv, activate it, and create another virtualenv within that ... but WHY? you are in a virtualenv already - and this guide does the same, but with a conda environment at the outer layer.

While the pip vs. conda problem can still appear - it's less problematic than creating a virtualenv within a virtual env - simply because it's unnecessary.

Conda setup is not that much different to the  [custom installation docs](https://www.freqtrade.io/en/latest/installation/#custom-installation) - except that you use `conda create -n freqtradeenv` instead of `python -m venv .env`.

---

> In due time, I intend to take a look at the existing documentation, because of the internal problems:

Feel free to do so, but while improving the documentation, keep in mind that it's already too large / long, and we should shorten it if possible, not make it even longer - as that will  lead to people losing interrest in reading it.

> there are fundamental deficiencies that do not allow the installation to be completed following the documentation.

You will not learn everything - but that's not the point, running the commands from top to bottom will get you a setup freqtrade docker instance running the sample strategy. 
We will however not go into detail on HOW to install docker (or how docker works). That's what the docker documentation is for.
Please open an issue about it if you are having problems with it so we can look where it goes wrong for you (this may avoid investing time trying to improve this, while it's not broken). I've just gone through these points again and they do work just fine and got the SampleStrategy setup correctly.

>docs/configuration.md
> It currently covers a broad spectrum of activities, but lacks the structure and transparency

Yes, everything that's a config option is "somehow" thrown in there ... I'm trying to clean it up from time to time, but i don't disagree that it's way too long, and some parts could be split out in subpages.

Reply:
##### conda vs venv:

my handling of their functions is that conda, would install libraries and clear path for dependencies - shortly : even ground and prepare territory where freqtrade will work. Remember, i had this dependency issue on 'random' linux distributions.

venv on the other had provide solution for this 

```bash
bash: freqtrade: command not found
```

and for any other purpose in fact i dont need venv

##### conda vs pip:

 "oh, i got a conda instance, so letme install dependency X via conda"
 this point could be clearly explained in the documentation, so someone who would go through it, would no had chance to miss it.
 
 ~ "You install other libraries on your own responsibility!"

##### Docker:
I cannot learn everything, but this i could. There gonna be day for that, so then we can talk. 
So far, without experience reference points, I see, that documentation step by step doesn't work for me.


##### Config:
This i have to form in my head. When accomplished, i would share it in separate thread

Reply:
> I see, that documentation step by step doesn't work for me.

I've just seen your issue (#4220). It's a Network problem with your docker daemon - you'll also have trouble downloading any other image. This is however not something we can or will explain in the freqtrade docs. Then it would be 1000nds of pages, should we want to cover these issues.

You can encounter the same with conda - but we'll also not explain how to properly configure a proxy, or fix why your network cable is broken. It's simply out of scope for this project's documentation.

Reply:
Let me dig thought it by myself,
if after homework i will have the question though, i would ask